pkg/gateway/multiparts/tracker.go

```diff
@@ -61,59 +94,31 @@ func (m *tracker) Get(ctx context.Context, uploadID string) (*MultipartUpload, e
 	if uploadID == "" {
 		return nil, ErrInvalidUploadID
 	}
-	res, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		var m MultipartUpload
-		if err := tx.Get(&m, `
-			SELECT upload_id, path, creation_date, physical_address, metadata, content_type 
-			FROM gateway_multiparts
-			WHERE upload_id = $1`,
-			uploadID); err != nil {
-			return nil, err
-		}
-		return &m, nil
-	})
+	data := &MultipartUploadData{}
+	err := m.store.GetMsg(ctx, []byte(uploadID), data)
 	if err != nil {
 		return nil, err
 	}
-	return res.(*MultipartUpload), nil
+	return &MultipartUpload{
+		UploadID:        data.UploadId,
+		Path:            data.Path,
+		CreationDate:    data.CreationDate.AsTime(),
+		PhysicalAddress: data.PhysicalAddress,
+		Metadata:        data.Metadata,
+		ContentType:     data.ContentType,
+	}, nil
 }
 
 func (m *tracker) Delete(ctx context.Context, uploadID string) error {
 	if uploadID == "" {
 		return ErrInvalidUploadID
 	}
-	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		res, err := tx.Exec(`DELETE FROM gateway_multiparts WHERE upload_id = $1`, uploadID)
-		if err != nil {
-			return nil, err
-		}
-		affected := res.RowsAffected()
-		if affected != 1 {
-			return nil, ErrMultipartUploadNotFound
+	if _, err := m.Get(ctx, uploadID); err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			return ErrMultipartUploadNotFound
 		}
-		return nil, nil
-	})
-	return err
-}
-
-func (m Metadata) Set(k, v string) {
-	m[strings.ToLower(k)] = v
-}
-
-func (m Metadata) Get(k string) string {
-	return m[strings.ToLower(k)]
-}
-func (m Metadata) Value() (driver.Value, error) {
-	return json.Marshal(m)
-}
-
-func (m *Metadata) Scan(src interface{}) error {
-	if src == nil {
-		return nil
+		return fmt.Errorf("failed on Get. key %s, %w", uploadID, err)
```

Comment:
golang convention for wrapping errors:
```suggestion
		return fmt.Errorf("failed on Get. key (%s): %w", uploadID, err)
```